### PR TITLE
E2E Tests: Run in Jetpack Docker

### PR DESCRIPTION
**Pull request by brbrr**
Switches E2E tests environment to use Jetpack Docker
I also commented out version check in plugin updater test, since I was not been able to make it work properly.

#### Jetpack product discussion

n/a
#### Does this pull request change what data or activity we track or use?

n/a
#### Testing instructions:

* Review the changes
* Make sure the CI is green

### CODE REVIEW

**Comment by adimoldovan** [comment]
LGTM in general, but I'm worried about pre-connection tests failing even after a re-run. It looks like during updater test the site is crashing. Maybe we should disable the updater test for now.

**Review by adimoldovan — CHANGES_REQUESTED** [review]
Tested some more and I cannot get the updater test to pass, both locally and in CI. Most times the test also leaves the env in a weird state making other pre-connection tests fail.
With updater test skipped all suites are passing.

File: projects/plugins/jetpack/tests/e2e/specs/pre-connection.test.js [hunk 1]
```diff
@@ -22,7 +22,7 @@ describe( 'Jetpack pre-connection', () => {
 
 	afterAll( async () => {
 		await execWpCommand(
-			`wp option update jetpack_private_options --format=json < ${ path.resolve(
+			`option update jetpack_private_options --format=json < ${ path.resolve(
```
Comment (adimoldovan) [inline projects/plugins/jetpack/tests/e2e/specs/pre-connection.test.js:25]: This breaks any following test that requires a connection. It adds the blog token back making the tests think there's an active connection but there isn't one. It doesn't seem to work in combination with `withCleanEnv()` in `beforeAll`. I think it should be removed entirely.

Comment (brbrr) [reply projects/plugins/jetpack/tests/e2e/specs/pre-connection.test.js]: Good catch! But I guess this is not related to this PR. Removed in 9e05ad5

**Comment by brbrr** [comment]
I think I managed to make the plugin updater test to work 🤷 🎉 🤷

**Review by adimoldovan — APPROVED** [review]
CI is green. Let's merge this!

**Pull request merged**